File: fact.go
package main

/*
   @@@ get_version: go version | awk '{print $3}' | sed 's/go//'
   @@@ before: go build fact.go
   @@@ instead: ./fact
   @@@ after: rm fact
   @@@ is_fast: 1
*/

import "fmt"

const (
	times         = 150000000
	fact16 uint64 = 20922789888000
)

// ----------------------------------------------
func fact(n uint64) uint64 {
	if n == 1 {
		return 1
	}
	return n * fact(n-1)
}

// ----------------------------------------------
func main() {
	ok := true

	for i := 0; i < times; i++ {
		ok = ok && fact(16) == fact16
	}

	if ok {
		fmt.Printf("Go finish %d - ok\n", times)
	} else {
		fmt.Printf("Go finish %d - fail\n", times)
	}
}

File: fact.java
/*
    @@@ get_version: docker run --rm -v "$PWD":/app -w /app openjdk:11-slim java -version 2>&1 | awk '$2 == "version" {print $3}' | sed 's/"//g'
    @@@ before: docker run --rm -v "$PWD":/app -w /app openjdk:11-slim javac fact.java
    @@@ instead: docker run --rm -v "$PWD":/app -w /app openjdk:11-slim java fact
    @@@ after: rm fact.class
    @@@ is_fast: 1
*/

public class fact {
    static long times = 300000000;
    static long fact16 = 20922789888000L;

    public static long fact(long n) {
        if (n == 1) return 1;
        return n * fact(n - 1);
    }

    public static void main(String[] args) {
        var ok = true;

        for (var i = 0L; i < times; i++) {
            ok = ok && fact(16) == fact16;
        }

        System.out.printf("Java finish %d - %s\n", times, ok ? "ok" : "fail");
    }
}
